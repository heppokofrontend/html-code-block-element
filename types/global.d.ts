import 'react';

declare global {
  // A type for the properties of a function component
  namespace React {
    interface CodeBlockHTMLAttributes<T> extends HTMLAttributes<T> {
      /** The accessible name of code block */
      label?: string | undefined;
      /** The Language name */
      language?: string | undefined;
      /** The flag to display the UI */
      controls?: boolean;
    }
  }

  // A type for JSX markup
  namespace JSX {
    interface IntrinsicElements {
      'code-block': React.DetailedHTMLProps<React.HTMLAttributes<HTMLPreElement>, HTMLPreElement> & {
        /** The accessible name of code block */
        label?: string | undefined;
        /** The Language name */
        language?: string | undefined;
        /** The flag to display the UI */
        controls?: boolean;
      }
    }
  }
}
