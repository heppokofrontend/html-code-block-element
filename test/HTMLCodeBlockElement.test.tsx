import * as React from 'react';
import '../types/global';
import renderer from 'react-test-renderer';

describe('React', () => {
  test('Simple', () => {
    const component = renderer.create(
        <code-block/>,
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('JSON', () => {
    const component = renderer.create(
        <code-block language="json">{
          `{
            "hoge": 100
          }`}
        </code-block>,
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('All', () => {
    const component = renderer.create(
        <code-block
          label="hoge"
          language="js"
          controls
        >
          <textarea><script>
            const hoge = 100;

            console.log(100);
          </script></textarea>
        </code-block>,
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
