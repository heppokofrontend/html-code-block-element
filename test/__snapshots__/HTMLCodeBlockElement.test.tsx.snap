// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React All 1`] = `
<code-block
  controls={true}
  label="hoge"
  language="js"
>
  <textarea>
    <script>
      const hoge = 100; console.log(100);
    </script>
  </textarea>
</code-block>
`;

exports[`React JSON 1`] = `
<code-block
  language="json"
>
  {
            "hoge": 100
          }
</code-block>
`;

exports[`React Simple 1`] = `<code-block />`;
